apps/syslog_drain.go

```diff
@@ -76,14 +76,13 @@ var _ = AppsDescribe("Logging", func() {
 func getSyslogDrainAddress(appName string) string {
 	var address []byte
 
-	Eventually(func() []byte {
+	Eventually(func() [][]byte {
 		re, err := regexp.Compile("ADDRESS: \\|(.*)\\|")
 		Expect(err).NotTo(HaveOccurred())
 
 		logs := cf.Cf("logs", appName, "--recent").Wait(Config.DefaultTimeoutDuration())
-		address = re.FindSubmatch(logs.Out.Contents())[1]
-		return address
-	}, Config.DefaultTimeoutDuration()).Should(Not(BeNil()))
+		return re.FindSubmatch(logs.Out.Contents())
+	}, Config.DefaultTimeoutDuration()).Should(HaveLen(2))
 
 	return string(address)
```

Comment:
Sure, but in the happy path, where do you take the matched string and assign it to `address` variable?

The return value of `FindSubmatch` is give to the `return` of the `Eventually` closure, which is used (correctly) against the `HaveLen` matcher. But it never gets assigned to `address`. I'm pretty sure you want something like:
```
var match [][]byte
Eventually(...
  ...
  logs := cf.Cf("logs", appName, "--recent").Wait(Config.DefaultTimeoutDuration())
  match = re.FindSubmatch(logs.Out.Contents())
  return match
}, ...).Should(HaveLen(2))
address = match[1]
```